src/main/scala/magellan/PolyLine.scala

```diff
@@ -103,6 +111,31 @@ class PolyLine(
     intersects
   }
 
+  def intersects(polygon:Polygon):Boolean = {
+    var startIndex = 0
+    var endIndex = 1
+    var intersects = false
+    val length = xcoordinates.size
+
+    breakable {
+
+      while(endIndex < length) {
+        val startX = xcoordinates(startIndex)
+        val startY = ycoordinates(startIndex)
+        val endX = xcoordinates(endIndex)
```

Comment:
@Charmatzis this code path has been refactored. please merge with master and use refactored code path

Reply:
Great, I had also in my mind that the we should refactor the PolyLine to look like Polygon.